### PR TITLE
Fix multi split for multilinestring

### PR DESCRIPTION
**Pull request by lbartoletti**
## Description

Split parts map tool, split only linestring (not multilinestring). 

I fix it with a simplification of the `QgsVectorLayerEditUtils::splitParts` method

Before
QGIS_cut_out_parts_tool mp4

After
split_multipart_after webm

### CODE REVIEW

**Comment by lbartoletti** [comment]
I'll add tests, after travis gives the go-ahead. I'm not sure about my change in  https://github.com/qgis/QGIS/pull/34510/files#diff-6707ca218add057b84fb80c314cf843aL839

File: src/core/qgsvectorlayereditutils.cpp [hunk 1]
```diff
@@ -475,35 +475,27 @@ QgsGeometry::OperationResult QgsVectorLayerEditUtils::splitParts( const QgsPoint
...
     QgsPointSequence topologyTestPoints;
     QgsGeometry featureGeom = feat.geometry();
     splitFunctionReturn = featureGeom.splitGeometry( splitLine, newGeometries, topologicalEditing, topologyTestPoints );
+
     if ( splitFunctionReturn == 0 )
     {
-      //add new parts
-      if ( !newGeometries.isEmpty() )
-        featureGeom.convertToMultiType();
+      QgsGeometry newGeom( newGeometries.at( 0 ) );
```
Comment (elpaso) [inline src/core/qgsvectorlayereditutils.cpp]: not sure if it's checked before, but this might crash if newGeometries is empty.

Comment (lbartoletti) [reply src/core/qgsvectorlayereditutils.cpp]: Normally with splitFunctionReturn, but I added a test if newGeometries is empty. Thanks.

File: src/core/geometry/qgsgeometry.cpp [hunk 2]
```diff
@@ -836,8 +836,8 @@ QgsGeometry::OperationResult QgsGeometry::splitGeometry( const QgsPointSequence
 
   if ( result == QgsGeometryEngine::Success )
   {
-    *this = newGeoms.takeAt( 0 );
-
+    if ( type() == QgsWkbTypes::PolygonGeometry && ! newGeoms.isEmpty() )
```
Comment (nyalldawson) [inline src/core/geometry/qgsgeometry.cpp]: why doesn't this apply to linestrings? it looks confusing to me

Comment (lbartoletti) [reply src/core/geometry/qgsgeometry.cpp]: No worries. It's really a WIP. I tried something with Travis (I don't always get the same results on my machine).

I think it's cleaner with my last commit. But I'm trouble to understand this fix...

**Comment by lbartoletti** [comment]
@nyalldawson @elpaso travis is green... I'll add tests, but I'm not really happy with this fix.

It breaks API with a new optional parameter which is revelant only for one case (splitParts... for multilinestring), and it doesn't make sense to duplicate splitGeometry methods.

So what do you think?

**Comment by lbartoletti** [comment]
@elpaso @nyalldawson the PR is ready for a final review. Thanks.

**Comment by lbartoletti** [comment]
ready to merge?

File: tests/src/app/CMakeLists.txt [hunk 3]
```diff
@@ -127,6 +127,9 @@ ADD_QGIS_TEST(maptoolmovefeaturetest testqgsmaptoolmovefeature.cpp)
 ADD_QGIS_TEST(maptoolellipsetest testqgsmaptoolellipse.cpp)
 ADD_QGIS_TEST(maptoolrectangletest testqgsmaptoolrectangle.cpp)
 ADD_QGIS_TEST(maptoolregularpolygontest testqgsmaptoolregularpolygon.cpp)
+IF(NOT WIN32)
+  ADD_QGIS_TEST(maptoolsplitpartstest testqgsmaptoolsplitparts.cpp)
+ENDIF(NOT WIN32)
```
[the remark this answers is not in the document]
Comment (lbartoletti) [reply tests/src/app/CMakeLists.txt]: No, azure is flaky :(

[the remark this answers is not in the document]
Comment (mhugo) [reply tests/src/app/CMakeLists.txt]: Ok. But, wouldn't it be better in the list of ignored tests here https://github.com/qgis/QGIS/blob/master/.ci/azure-pipelines/azure-pipelines.yml#L4 ?

[the remark this answers is not in the document]
Comment (lbartoletti) [reply tests/src/app/CMakeLists.txt]: 😨 of course... :)

File: python/core/auto_generated/geometry/qgsgeometry.sip.in [hunk 4]
```diff
@@ -839,29 +839,32 @@ Rotate this geometry around the Z axis
...
+    OperationResult splitGeometry( const QgsPointSequence &splitLine, QVector<QgsGeometry> &newGeometries /Out/, bool topological, QgsPointSequence &topologyTestPoints /Out/, bool splitFeature = true );
 %Docstring
 Splits this geometry according to a given line.
 
 :param splitLine: the line that splits the geometry
 \param[out] newGeometries list of new geometries that have been created with the split
 :param topological: ``True`` if topological editing is enabled
 \param[out] topologyTestPoints points that need to be tested for topological completeness in the dataset
+:param splitFeature: TODO DESCRIPTION: can we really add this parameter to
+                     fix this bug?
```
[the remark this answers is not in the document]
Comment (lbartoletti) [reply python/core/auto_generated/geometry/qgsgeometry.sip.in]: yep; done

**Review by mhugo — APPROVED** [review]

**Pull request merged**